--- a/drivers/dahdi/Kbuild
+++ b/drivers/dahdi/Kbuild
@@ -52,7 +52,7 @@ obj-$(DAHDI_BUILD_ALL)$(CONFIG_DAHDI_WCB
 
 endif
 
-obj-$(DAHDI_BUILD_ALL)$(CONFIG_DAHDI_XPP)		+= xpp/
+#obj-$(DAHDI_BUILD_ALL)$(CONFIG_DAHDI_XPP)		+= xpp/
 
 obj-$(DAHDI_BUILD_ALL)$(CONFIG_DAHDI_ECHOCAN_JPAH)	+= dahdi_echocan_jpah.o
 obj-$(DAHDI_BUILD_ALL)$(CONFIG_DAHDI_ECHOCAN_STEVE)	+= dahdi_echocan_sec.o
--- a/drivers/dahdi/Kconfig
+++ b/drivers/dahdi/Kconfig
@@ -245,4 +245,4 @@ config DAHDI_HFCS
 
 	  If unsure, say Y.
 
-source "drivers/dahdi/xpp/Kconfig"
+#source "drivers/dahdi/xpp/Kconfig"
--- a/drivers/dahdi/xpp/Kconfig
+++ b/drivers/dahdi/xpp/Kconfig
@@ -5,7 +5,7 @@
 menuconfig DAHDI_XPP
 	tristate "Xorcom Astribank Support"
 	depends on DAHDI
-	default DAHDI
+	default n
 	---help---
 	  Infrastructure support for Xorcom Astribank products.
 
