--- a/drivers/dahdi/Kbuild
+++ b/drivers/dahdi/Kbuild
@@ -69,7 +69,11 @@ ifneq (,$(wildcard $(src)/../staging/ech
 obj-m += dahdi_echocan_oslec.o
 endif
 
-CFLAGS_MODULE += -I$(DAHDI_INCLUDE) -I$(src) -Wno-format-truncation
+ifeq ($(CC),clang)
+  CFLAGS_MODULE += -I$(DAHDI_INCLUDE) -I$(src)
+else
+  CFLAGS_MODULE += -I$(DAHDI_INCLUDE) -I$(src) -Wno-format-truncation
+endif
 
 # fix typo present in CentOS and RHEL 2.6.9 kernels
 BAD_KERNELS_VERS := 22 34 34.0.1 34.0.2
